[Reland] Refactor stack_trace preservation for node meta preservation (#90803)

SherlockNoMad opened the pull request:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/90803
Approved by: https://github.com/jerryzh168, https://github.com/albanD
ghstack-source-id: 5848cca08ef5d6f8868f4f79d8bc29711e9a52c2

Fixes #ISSUE_NUMBER


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

albanD reviewed (commented):
What's the difference with the original PR?

SherlockNoMad committed:
[Reland] Refactor stack_trace preservation for node meta preservation (#90803) (#90803)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/90803
Approved by: https://github.com/jerryzh168, https://github.com/albanD
ghstack-source-id: 5848cca08ef5d6f8868f4f79d8bc29711e9a52c2

Fixes #ISSUE_NUMBER

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Pull Request resolved: https://github.com/pytorch/pytorch/pull/92400

Reviewed By: qihqi, jerryzh168

Differential Revision: D42565536

Pulled By: SherlockNoMad

fbshipit-source-id: 455060a24f5906467d74ee848731f9a2aca31864
